18_regexp_lab/01.py

```diff
@@ -1,12 +1,35 @@
-"""
-Write a program that reads data
-from property files.
-Each line in the file can either be:
-    An empty line
-    A comment line (Start with #)
-    A property line (of the form key = value)
-
-Write a program that takes a property file name and key
-as command line arguments and prints the requested value
-"""
-
+#!/usr/bin/env python
+
+__author__ = 'Eabel'
+
+import sys
+import os
+import argparse
+import re
+
+
+########################################################################################################################
+#
+#
+def main():
+    parser = argparse.ArgumentParser(description='This program receive config file path and varible name and prints the varible value')
+    
+    parser.add_argument("config_path", help="path to config file")
+    parser.add_argument("varible_name", help="varible name")
+    args = parser.parse_args()
+
+    try:
+        regex_str = '\s*' + args.varible_name + '\s*=\s*(.*)'

```

Comment:
Sure I thinks so too, but it was regex practice so... 

Sent from my iPhone

> On 12 Jul 2016, at 21:52, ynonp <notifications@github.com> wrote:
> 
> In 18_regexp_lab/01.py:
> 
> > +import argparse
> > +import re
> > +
> > +
> > +########################################################################################################################
> > +#
> > +#
> > +def main():
> > +    parser = argparse.ArgumentParser(description='This program receive config file path and varible name and prints the varible value')
> > +    
> > +    parser.add_argument("config_path", help="path to config file")
> > +    parser.add_argument("varible_name", help="varible name")
> > +    args = parser.parse_args()
> > +
> > +    try:
> > +        regex_str = '\s*' + args.varible_name + '\s*=\s*(.*)'
> Better not to use external input as regular expression. Many characters have special meaning in regexp context (such as: $, ^, \, *, ., etc.)
> 
> You can get better results here by reading the file into a dictionary (split every line and use the values as key/value pair), and then search the dictionary for the required value.
> 
> Also beware of '#' lines which are comments
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>